drivers/extend/gpio_redir.c

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright (C) 2018 Acutam Automation, LLC
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_extend_gpio
+ *
+ * @{
+ *
+ * @file
+ * @brief       GPIO extension redirection functions
+ *
+ * @author      Matthew Blue <matthew.blue.neuro@gmail.com>
+ *
+ * @}
+ */
+
```

Comment:
Please add `#if  MODULE_EXTEND_GPIO ` here.

Otherwise you get compilation errors if module `extend` is enabled but not pseudomodule `extend_gpio`, e.g., because pseudomodule `extend_adc` is enabled.

Reply:
I feel like there should be a better way of accomplishing this, because this is not needed for e.g. periph, but I am not sure what it is. I have fixed it as you suggested for now.

Reply:
The reason is that `extend` module is compiled if any `extend_*` pseudomodule is enabled because of the dependency in `drivers/Makefile.dep`, e.g.,
```
ifneq (,$(filter extend_adc,$(USEMODULE)))
  USEMODULE += extend
endif
```
and all `*_extend` files reside in `drivers/extend` directory.

So, make changes to the `drivers/extend` directory if any 'extend_*` module is enabled and tries to compile all .c files found in this directory including `gpio_extend.c` independent on whether `extend_gpio` is enabled or not.

We could solve it to add a separate directory for each `extend_*` module but I would
prefer to let all `*_extend` files in one directory instead having a separate directory and to solve it with `#if MODULE_* ... #endif` wrappers.